### PR TITLE
10-10-Make a console out

### PR DESCRIPTION
**Pull request by PawelPaziewski**
I wait for your code review.

### CODE REVIEW

File: src/main/java/ConsoleOut/ColorCoder.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package ConsoleOut;
+
+public enum ColorCoder {
+    BLACK("\u001B[30m"), RED("\u001B[31m"), GREEN("\u001B[32m"), YELLOW("\u001B[33m"), BLUE("\u001B[34m"), PURPLE("\u001B[35m"), WHITE("\u001B[37m"),CYAN("\u001B[36m");
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/ColorCoder.java]: Formatting! (I forgot to mention that we should use google formatter)

File: src/main/java/ConsoleOut/ConsoleOut.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import Main.History;
+
+public class ConsoleOut implements UserWriter {
+    private TextWriter textWriter;
+    private WaysWriter waysWriter;
+    private int delay;
+    private String color;
+    public ConsoleOut(int delay, String color) {
+        this.textWriter = new TextWriter();
+        this.waysWriter = new WaysWriter();
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/ConsoleOut.java]: You can't test this. We shouldn't use new Keyword.

File: src/main/java/ConsoleOut/ConsoleOut.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        this.color= ColorCoder.getColorCode(color);
+        this.delay = delay;
+    }
+    public void writeHistory(History history){
+        textWriter.Write(history.getHistory(), delay, color);
+        waysWriter.Write(history.getWays(), delay, color);
+    }
+}
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/ConsoleOut.java]: This class has 19 lines of code, text writer has 15 and ways writer has 19. I think that this granularity is not good in here. Methods from text and ways writers can be in this class ( private methods).

File: src/main/java/ConsoleOut/ConsoleOut.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    private String color;
+    public ConsoleOut(int delay, String color) {
+        this.textWriter = new TextWriter();
+        this.waysWriter = new WaysWriter();
+        this.color= ColorCoder.getColorCode(color);
+        this.delay = delay;
+    }
+    public void writeHistory(History history){
+        textWriter.Write(history.getHistory(), delay, color);
+        waysWriter.Write(history.getWays(), delay, color);
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/ConsoleOut.java]: Methods names should be in lowerCase

File: src/main/java/ConsoleOut/WaysWriter.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package ConsoleOut;
+
+import Main.Way;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+public class WaysWriter {
+    public void Write(Map<Character, Way> ways, int delay, String color){
+        Set<Entry<Character, Way>> entrySet=ways.entrySet();
+        for(Entry<Character, Way> entry:entrySet){
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/WaysWriter.java]: Unnecessary Set. You can iterate directly on ways.entrySet();

**Review by Dandoo1212 — CHANGES_REQUESTED** [review]

File: src/main/java/ConsoleOut/ConsoleOut.java [hunk 6]
```diff
@@ -1,19 +1,45 @@
...
-        textWriter.Write(history.getHistory(), delay, color);
-        waysWriter.Write(history.getWays(), delay, color);
+
+    public void writeHistory(History history) {
+        writeStory(history.getHistory());
+        writeWays(history.getWays());
+    }
+
+    void writeStory(String text) {
+        for (char c : text.toCharArray()) {
```
Comment (Dandoo1212) [inline src/main/java/ConsoleOut/ConsoleOut.java]: If You want, try to learn something about streams in Java (collection streams) and replace this for each loop with a stream, or a for each with lamba ;)

File: src/main/java/pl/rpg/console/ConsoleOut.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+package pl.rpg.console;
+
+
+import pl.rpg.game.History;
+import pl.rpg.game.UserWriter;
+import pl.rpg.game.Way;
+
+import java.util.Map;
+
+public class ConsoleOut implements UserWriter {
```
Comment (Dandoo1212) [inline src/main/java/pl/rpg/console/ConsoleOut.java:10]: Does your code compile?

Comment (PawelPaziewski) [reply src/main/java/pl/rpg/console/ConsoleOut.java:10]: Yes, it is

**Review by Dandoo1212 — APPROVED** [review]

**Pull request merged**